### ISSUE TITLE
[Question] Read card number from mobib

### ISSUE THREAD

**Issue by lorenc-tomasz** [flamingokweker/Android-NFC-Mobib-Reader#1]
Hi,
I have mobib card and I would like to read it's card number that should be inside the holder data. But when I try to select APDU I'm getting 6E 00 answer and that's it. Any idea how to read card number from mobib / pcard card ?

**Comment by flamingokweker** [comment]
Hi Thomasz,

What is the complete command that you send to the card when you try to
select the mobib application?

6E00 normally means that your CLA byte is not correct. (=the first byte of
your apdu)

I send 00A4000002206800 and my card seems to respond fine.

Regards,
Ken

**Comment by lorenc-tomasz** [comment]
Thanks for quick answer. I was playing with instruction I found here: https://github.com/zoobab/mobib-extractor/blob/master/pymobibACR38.py 

I have tried your, and got 6B:00 as answer. I tried app SNCB MoBIB available in store and it reads the card number and expiration date as it should.

**Comment by lorenc-tomasz** [comment]
OK. It seems that I managed to read it. 

1. Select APDU: xxxx
2. Read holder: xxxx

Holder:
```
# - bytes 0-1 : unknown data
# - bytes 2-11 : the card number
# - bytes 12-20 : unknown data
# - bytes 21-24 : the birthday
# - bytes 24-28 : the begining of the name
```

3. Decode card number in python3: 

```python
numero = ['0','1','2','3','4','5','6','7','8','9']

## Hexadecimal to binary
def hex_to_bin(h):
	res = ''
	for i in range(0,len(h)):
		if h[i] in numero:
			a = int(h[i]) 
		else:
			if h[i] == "A":
				a = int(10)
			if h[i] == "B":
				a = int(11)
			if h[i] == "C":
				a = int(12)
			if h[i] == "D":
				a = int(13)
			if h[i] == "E":
				a = int(14)
			if h[i] == "F":
				a = int(15)
		for j in (3,2,1,0):
			if a-pow(2,j) >= 0:
				res = res + "1"
				a = a - pow(2,j)
			else:
				res = res + "0"
	return res

## Binary to number with offset t (on 4 bits)
def bin_to_number_dec (b, t):
	res = ''
	a = int(0)
	if len(b)%4 == 0:
		r = 4
	else:
		r = len(b)%4
	for i in range(0+t,len(b)-(r+t), 4):
		a = int(b[i])*8 + int(b[i+1])*4 + int(b[i+2])*2 + int(b[i+3])
		if a > 9:
			res = res + "x"
		else:
			res = res + str(a)
	return res	

holder1 = hex_string.split()
hexa_card = ''.join(holder1[2:12]) + holder1[12][0]
num_card = bin_to_number_dec(hex_to_bin(hexa_card), 2)
print("Card number: %s\n" % num_card)
```

`hex_string` is response from reading the holder record. 

Maybe it will be useful for someone else

**Comment by flamingokweker** [comment]
Hi,

To be honest, I haven't really looked at or maintained the mobib reader app
I created for quite some time.

It's been 7 years according to github :)

It works with my 'anonymous' reloadable mobib card I got from MIVB. It has
never been tested with other card types.

If you have tools to spy the official mobib app, then that's probably your
best bet.

If you're stuck on something, I can always try to help/answer.

Ken.
